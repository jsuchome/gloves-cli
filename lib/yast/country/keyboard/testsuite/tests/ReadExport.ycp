/**
 * tests for Keyboard.ycp autoyast functions
 * Author:	jsuchome@suse.cz
 * $Id: ReadExport.ycp 57028 2009-04-29 10:58:09Z lslezak $
 */
{
    // testedfiles: Keyboard.ycp

    import "Testsuite";

    map READ = $[
	"sysconfig" : $[
	    "keyboard" : $[
		"YAST_KEYBOARD"	: "english-us,pc104",
	    ],
	],
	"target" : $[
	    "size"		: 1,
	    "dir"		: [],
	    "yast2"		: $[
		"english-us"	: [
		    "English (US)",
		    $[
			"pc104": $[
				"ncurses": "us.map.gz",
			],
		    ]
		],
	    ],
	    "tmpdir"		: "/tmp",
	],
	"probe"	: $[
	    "architecture"	: "i386"
	],
    ];

    map E	= $[
	"target" : $[
	    "bash_output"	: $[],
	],
    ];

    import "Mode";

    // let's simulate Mode::config
    Mode::SetMode ("autoinst_config");

    Testsuite::Init ([READ,$[],E], nil);

    import "Keyboard";

    Testsuite::Test (``(Keyboard::Restore ()), [READ,$[],E], nil);

    Testsuite::Test (``(Keyboard::Read ()), [], nil);

    Testsuite::Test (``(Keyboard::Export ()), [], nil);

    Testsuite::Test (``(Keyboard::Summary ()), [], nil);

    Testsuite::Test (``(Keyboard::Modified ()), [], nil);
}
