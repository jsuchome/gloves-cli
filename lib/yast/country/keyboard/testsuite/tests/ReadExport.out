Read	.target.yast2 "keyboard_raw.ycp" $["english-us":["English (US)", $["pc104":$["ncurses":"us.map.gz"]]]]
Return	true
Return	true
Return	$["keymap":"english-us"]
Return	<ul><li>Current Keyboard Layout: English (US)</li></ul>
Return	false
